multimodal/src/autogluon/multimodal/optimization/lit_mmdet.py

```diff
@@ -219,29 +199,48 @@ def log_step_results(self, losses):
             self.log(f"step/{loss_name}", losses[key])
 
     def training_step(self, batch, batch_idx):
-        loss, log_vars = self._training_step(batch, batch_idx)
+        loss, log_vars = self._loss_step(
+            img=batch[self.input_data_key]["img"],
+            img_metas=batch[self.input_data_key]["img_metas"],
+            gt_bboxes=batch[self.input_data_key]["gt_bboxes"],
+            gt_labels=batch[self.input_data_key]["gt_labels"],
+        )
+        # log step losses
+        self.log_step_results(log_vars)
         return loss
 
     def validation_step(self, batch, batch_idx, dataloader_idx=0):
-        self.evaluate(batch, "val")
+        if self.use_loss:
+            val_loss, log_vars = self._loss_step(
+                img=batch[self.input_data_key]["img"],
+                img_metas=batch[self.input_data_key]["img_metas"],
+                gt_bboxes=batch[self.input_data_key]["gt_bboxes"],
+                gt_labels=batch[self.input_data_key]["gt_labels"],
+            )
+            self.validation_metric.update(val_loss)
+        else:
+            self.evaluate(batch, "val")
 
     def validation_epoch_end(self, validation_step_outputs):
-        # TODO: add mAP/mAR_per_class
-        mAPs = {"val_" + k: v for k, v in self.validation_metric.compute().items()}
-        mAPs["val_mAP"] = mAPs["val_map"]
-        self.print(mAPs)
-        self.log_dict(mAPs, sync_dist=True)
+        val_result = self.validation_metric.compute()
+        if self.use_loss:
+            self.print("val_loss: %.2f" % val_result.item())
```

Comment:
I think we may always use `self.log()` and avoid `self.print()`. `self.print()` may only be used when debugging the code.

Reply:
Done.